src/clang_parser.h

```diff
@@ -3,6 +3,9 @@
 #include <clang-c/Index.h>
 #include "bpftrace.h"
 
+#define ASM_GOTO_WORKAROUND_H "asm_goto_workaround.h"
```

Comment:
Actually, screw preprocessor safety - we could go the C++ way and use a constant string inside the bpftrace namespace

Reply:
Nice! I'll update the PR with that.

Reply:
the change works in my config, thanks

Reply:
> we could go the C++ way and use a constant string inside the bpftrace namespace

CXUnsavedFile expects a `char*`, and not a `std::string`

```
/vagrant/src/clang_parser.cpp: In member function ‘bool bpftrace::ClangParser::parse(bpftrace::ast::Program*, bpftrace::BPFtrace&, std::vector<std::__cxx11::basic_string<char> >)’:
/vagrant/src/clang_parser.cpp:331:3: error: cannot convert ‘std::__cxx11::basic_string<char>’ to ‘const char*’ in initialization
   };
   ^
/vagrant/src/clang_parser.cpp: In member function ‘bool bpftrace::ClangParser::parse(bpftrace::ast::Program*, bpftrace::BPFtrace&, std::vector<std::__cxx11::basic_string<char> >)’:
/vagrant/src/clang_parser.cpp:331:3: error: cannot convert ‘std::__cxx11::basic_string<char>’ to ‘const char*’ in initialization
   };
   ^
```

For now let's just duplicate the filename in main.cpp and clang_parser.cpp. We can improve this later.

Reply:
How about `const char *asm_goto_workaround_h = "asm_goto_workaround.h"`? Then we can put that inside the bpftrace namespace and not have to use #defines

Reply:
We're building up quite a lot of technical debt in bpftrace so I don't want to keep putting off improvements until later - we should try and do stuff right the first time

Reply:
And use as:

```c++
    .Filename = "/bpftrace/include/" + asm_goto_workaround_filename,
```

? Doesn't seem to work either:

```
/vagrant/src/clang_parser.cpp: In member function ‘bool bpftrace::ClangParser::parse(bpftrace::ast::Program*, bpftrace::BPFtrace&, std::vector<std::__cxx11::basic_string<char> >)’:                                                                                                                                                                                                                                                         
/vagrant/src/clang_parser.cpp:331:3: error: cannot convert ‘std::__cxx11::basic_string<char>’ to ‘const char*’ in initialization                                                                                                                                                                                                                                                                                                             
   };                                                                                                                                                                                                                                                                                                                                                                                                                                        

```

Reply:
Ah forgot about the `+`... damn it. We could concatenate them in a `std::string` and then store the `.c_str()` in `.Filename`, but it feels messy. Guess the #define will have to do then.

We can still get rid of `ASM_GOTO_WORKAROUND_PATH` though - see my other comment